imi/src/main/java/org/motechproject/nms/imi/service/impl/CdrFileServiceImpl.java

```diff
@@ -724,6 +731,8 @@ public void verifyDetailFileChecksumAndCount(CdrFileNotificationRequest request)
                 null
         ));
 
+        // TODO: is there a better place/way to trigger this?
+        cleanOldCallRecords();
```

Comment:
yep, I would send that as a motech event so some other node can handle it asynchronously